test/integration/util/util.go

```diff
@@ -553,14 +553,14 @@ func InitTestAPIServer(t *testing.T, nsPrefix string, admission admission.Interf
 }
 
 // WaitForSchedulerCacheCleanup waits for cleanup of scheduler's cache to complete
-func WaitForSchedulerCacheCleanup(sched *scheduler.Scheduler, t *testing.T) {
-	schedulerCacheIsEmpty := func() (bool, error) {
+func WaitForSchedulerCacheCleanup(ctx context.Context, sched *scheduler.Scheduler, t *testing.T) {
```

Comment:
Same here: `tCtx` replaces `ctx` and `t`.

Reply:
There is another inconsistency in these helper functions:
- WaitForSchedulerCacheCleanup does not return an error and instead records a test failure via t.
- WaitForNodeTaints returns an error and the caller must add a `expect no error` check. I've found tests elsewhere where the error checking was missing. We now have a non-blocking linter check for it, but still...

Other tests have abused such "wait + return error" functions in situations where polling was expected to time out. Usually, *any* error (not just the timeout error) made the test pass. A "consistently" check is better.

My preference nowadays is to not return errors in functions which are meant to succeed for a test. If the caller needs to add some details to the test failure, they can do with `WithStep`.

Traditional:
```
err := WaitForNodeTaints(ctx, ...)
if err != nil {
    t.Fatalf("waiting for XYZ: %v", err)
}
```

Alternative:
```
WaitForNodeTaints(ktesting.WithStep(tCtx, "waiting for XYZ", ...)
```

For stack unwinding to point to the test code, `WaitForNodeTaints` would need to call `tCtx.Helper()`.